### PR TITLE
BaseCheckpointCallback

### PR DESCRIPTION
**Pull request by TezRomacH**

### CODE REVIEW

File: catalyst/dl/callbacks/checkpoint.py [hunk 1]
```diff
@@ -9,19 +9,43 @@
...
+    def __init__(
+        self,
+        metric_filename: str
+    ):
+        self.metric_filename = metric_filename
+
+    def get_metrics(self, **kwargs) -> Dict:
+        pass
+
+    def remove_checkpoints(self, **kwargs) -> None:
```
Comment (Scitator) [inline catalyst/dl/callbacks/checkpoint.py]: lets rename to `remove_checkpoint`

File: catalyst/dl/callbacks/checkpoint.py [hunk 2]
```diff
@@ -9,19 +9,43 @@
...
 
 
-class CheckpointCallback(Callback):
+class BaseCheckpointCallback(Callback):
+    """
+    Base class for all checkpoint callbacks
+    """
+    def __init__(
+        self,
+        metric_filename: str
```
Comment (Scitator) [inline catalyst/dl/callbacks/checkpoint.py]: lets add some default value here

File: catalyst/dl/callbacks/checkpoint.py [hunk 3]
```diff
@@ -9,19 +9,43 @@
...
+        save_n_best: int = 3,
+        resume: str = None,
+        resume_dir: str = None
     ):
         """
         Args:
             save_n_best: number of best checkpoint to keep
             resume: path to checkpoint to load and initialize runner state
         """
+        super().__init__(metric_filename="_metrics.json")
```
Comment (Scitator) [inline catalyst/dl/callbacks/checkpoint.py]: we need to add `metric_filename` to `CheckpointCallback` init args

File: catalyst/dl/callbacks/checkpoint.py [hunk 4]
```diff
@@ -74,27 +122,10 @@ def save_checkpoint(
...
-            for (filepath, _, valid_metric) in self.top_best_metrics
-        ]
-        best_valid_metrics = checkpoints[0][1]
-        metrics = OrderedDict(
-            [("best", best_valid_metrics)] +
-            checkpoints +
-            [("last", valid_metrics)]
-        )
-        safitty.save(metrics, f"{logdir}/checkpoints/_metrics.json")
+        metrics = self.get_metrics(valid_metrics)
```
Comment (Scitator) [inline catalyst/dl/callbacks/checkpoint.py]: maybe, `get_metrics_file`?

File: catalyst/dl/callbacks/checkpoint.py [hunk 5]
```diff
@@ -9,19 +9,43 @@
 from catalyst.dl.core import Callback, RunnerState
 
 
-class CheckpointCallback(Callback):
+class BaseCheckpointCallback(Callback):
```
Comment (Scitator) [inline catalyst/dl/callbacks/checkpoint.py:12]: I think, `BaseCheckpointCallback` should have:
```
save_checkpoint
remove_checkpoint or truncate_worst_checkpoint ?
get_metrics_file
save_metrics_file
```
?

**Review by Scitator — CHANGES_REQUESTED** [review]

File: catalyst/dl/callbacks/checkpoint.py (hunk 4 above)
Comment (TezRomacH) [reply catalyst/dl/callbacks/checkpoint.py]: it's not a file, it's actually a dict

File: catalyst/dl/callbacks/checkpoint.py (hunk 5 above)
Comment (TezRomacH) [reply catalyst/dl/callbacks/checkpoint.py:12]: without `get_metrics_file`

**Pull request merged**